File: frontend/src/microservices/flight/objects/FlightListPlannerResponse.ts
// 本文件定义 flight 模块的 `FlightListPlannerResponse`，作为列表响应数据并提供 JSON 编解码。

import type { FlightPlannerResponse } from "./FlightPlannerResponse"

export type FlightListPlannerResponse = {
  flights: FlightPlannerResponse[]
}

export const flightListPlannerResponseFromJson = (json: string): FlightListPlannerResponse =>
  JSON.parse(json) as FlightListPlannerResponse

export const flightListPlannerResponseToJson = (value: FlightListPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/FlightListPlannerResponse.scala
// FlightListPlannerResponse defines the flight search planner response model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightListPlannerResponse(flights: List[FlightPlannerResponse])

object FlightListPlannerResponse:
  given sourceEncoder: Encoder[FlightListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[FlightListPlannerResponse] = deriveDecoder

